Source/WebCore/html/PDFDocument.cpp

```diff
@@ -181,32 +181,48 @@ void PDFDocument::finishedParsing()
         sendPDFArrayBuffer();
 }
 
-void PDFDocument::sendPDFArrayBuffer()
+void PDFDocument::postMessageToIframe(const String& name, JSC::JSObject* data)
 {
-    using namespace JSC;
-
-    auto* frame = m_iframe->contentFrame();
-    // FIXME: https://bugs.webkit.org/show_bug.cgi?id=236668 - Use postMessage
-    auto openFunction = frame->script().executeScriptIgnoringException("PDFJSContentScript.open"_s).getObject();
-
     auto globalObject = this->globalObject();
     auto& vm = globalObject->vm();
+    JSC::JSLockHolder lock(vm);
+
+    JSC::JSObject* message = constructEmptyObject(globalObject);
+    message->putDirect(vm, JSC::Identifier::fromString(vm, "message"_s), JSC::jsNontrivialString(vm, name));
+    if (data)
+        message->putDirect(vm, JSC::Identifier::fromString(vm, "data"_s), data);
+
+    auto* contentWindow = m_iframe->contentFrame()->window();
+    auto* contentWindowGlobalObject = m_iframe->contentDocument()->globalObject();
+
+    if (!data) {
+        auto returnValue = contentWindow->postMessage(*contentWindowGlobalObject, *contentWindow, message, WindowPostMessageOptions { });
+        if (returnValue.hasException())
+            returnValue.releaseException();
+    } else {
+        JSC::Strong<JSC::JSObject> dataRef { vm, data };
+
+        auto returnValue = contentWindow->postMessage(*contentWindowGlobalObject, *contentWindow, message, "*"_s, Vector { dataRef });
+        if (returnValue.hasException())
+            returnValue.releaseException();
+    }
```

Comment:
Nit: I think it'd be a little cleaner to not duplicate the exception logic, maybe something like:

```
    WindowPostMessageOptions options;
    if (data)
        options = WindowPostMessageOptions { "*"_s, JSC::Strong<JSC::JSObject> { vm, data }  };
    auto returnValue = contentWindow->postMessage(*contentWindowGlobalObject, *contentWindow, message, WTFMove(options));
    if (returnValue.hasException())
        returnValue.releaseException();
 ```

Reply:
That's much cleaner, thanks!